lib/access-context.js

```diff
@@ -88,7 +88,9 @@ function AccessContext(context) {
   var token = this.accessToken || {};
 
   if (token.userId) {
-    this.addPrincipal(Principal.USER, token.userId);
+    var userPrincipalType =
+      (this.accessToken && this.accessToken.principalType) || Principal.USER;
```

Comment:
I think this can be shortened by using `token` instead of `accessToken`?

Also please use `const` instead of `var` in new code - see http://loopback.io/doc/en/contrib/style-guide.html#variable-declarations

```ts
const userPrincipalType = token.principalType || Principal.USER;
```

Reply:
I'll complete the patch soon.